infrastructure/network/connmanager/connmanager.go

```diff
@@ -170,6 +171,8 @@ func (c *ConnectionManager) connectionExists(addressString string) bool {
 }
 
 func (c *ConnectionManager) isPermanent(addressString string) bool {
+	c.connectionRequestsLock.Lock()
```

Comment:
I think it's better to just add this in the exported function that calls it (to avoid unexpected deadlocks)